### PR TITLE
[TIMOB-24425] iOS/Android: Add touchEnabled() to Overlays

### PR DESCRIPTION
**Pull request by m1ga**
https://jira.appcelerator.org/browse/TIMOB-24425

### CODE REVIEW

File: apidoc/Polyline.yml [hunk 1]
```diff
@@ -31,6 +31,11 @@ properties:
     summary: Line width in pixels to use when drawing the polyline.
     type: Number
     default: 10
+    
+  - name: touchEnabled
+    summary: Determines whether view should receive touch events.
+    type: Boolean
+    default: true
```
Comment (hansemannn) [inline apidoc/Polyline.yml]: `since: "6.1.0"`

File: android/src/ti/map/PolylineProxy.java [hunk 2]
```diff
@@ -25,29 +25,28 @@
...
 	private Polyline polyline;
+	private boolean clickable;
 
 	private static final int MSG_FIRST_ID = KrollProxy.MSG_LAST_ID + 1;
 
 	private static final int MSG_SET_POINTS 	  = MSG_FIRST_ID + 499;
 	private static final int MSG_SET_STROKE_COLOR = MSG_FIRST_ID + 501;
 	private static final int MSG_SET_STROKE_WIDTH = MSG_FIRST_ID + 502;
 	private static final int MSG_SET_ZINDEX 	  = MSG_FIRST_ID + 503;
+	private static final int MSG_SET_TOUCHENABLED = MSG_FIRST_ID + 504;
```
Comment (hansemannn) [inline android/src/ti/map/PolylineProxy.java]: `MSG_SET_TOUCH_ENABLED `

File: apidoc/Polyline.yml (hunk 1 above)
Comment (hansemannn) [inline apidoc/Polyline.yml]: I think iOS does not support disabling it manually, so `platforms: [android]`

Comment (m1ga) [inline apidoc/Polyline.yml]: would be good to have that too. If you create a polyring (path around an area) with annoations in the middle you'll get weird results clicking on an annotation. E.g. you can't deselect a selected annotation by clicking inside the ring on the map

Comment (hansemannn) [inline apidoc/Polyline.yml]: Actually iOS can solve this very easy, also for polygon and circle as well. If we do this, we should do it for both platforms and all overlay-types. Can you do that on Android? Then I'll add my iOS changes here.

Comment (m1ga) [inline apidoc/Polyline.yml]: sure

**Comment by TonyCat** [comment]
Can a title be added for polygons and polylines?

**Comment by m1ga** [comment]
Perhaps, but that is not part of this PR :)

**Comment by yozef** [comment]
Map overlays on Android have huge performance issues due to the click on
them.

It loops through each overlay (imagine 100s of overlays), so it freezes on
Android, it would be ideal to select which of the overlays can be
touchEnabled.

**Comment by yozef** [comment]
Map overlays on Android have huge performance issues due to the click on them.

It loops through each overlay (imagine 100s of overlays), so it freezes on Android, it would be ideal to select which of the overlays can be touchEnabled otherwise they would be omitted.

**Comment by hansemannn** [comment]
Thank you @m1ga! @yozef But isn't that what this PR does? To address exactly that issue. Will push iOS later today. Unfortunately 6.1.0 is closed by now, so it will go into 6.2.0 but still available earlier through Github for the ones who need it.

**Comment by hansemannn** [comment]
@m1ga Pushed to your branch, feel free to check it out.

**Comment by yozef** [comment]
Ah sorry then, I didn't read up on the code details :) this PR is awesome
then!

**Comment by hansemannn** [comment]
CR + FT **passed**, PR **approved**!

**Pull request merged**